solution/index.js

```diff
@@ -0,0 +1,356 @@
+    const draggable = document.querySelector('.dragging') // only one el with class 'dragging' at a time
+    if (immediatelyBelowElement == 'null') {
+      ul.append(immediatelyBelowElement)
+    } else if (ul.className === 'trash-container') {
+      if (draggable) draggable.remove()
+    } else {
+      ul.insertBefore(draggable, immediatelyBelowElement)
+    }
+    updateParsedTasksAndSetLocalStorage()
+  })
+})
+
+function getDragAfterElement(ul, yCoordinate) {
+  // Y axis of where your mouse is
+  const draggableElements = [...ul.querySelectorAll('.task:not(.dragging)')] // get every draggable but the one your'e currently dragging
+  return draggableElements.reduce(
+    (closestElement, task) => {
+      const box = task.getBoundingClientRect() // rectangle of <li>
+      const offset = yCoordinate - box.top - box.height / 2 // positive if above the half line of box, negative otherwise
+      if (offset < 0 && offset > closestElement.offset) {
+        // takes the element with the losest POSITIVE offset
+        return { offset: offset, element: task }
+      } else {
+        return closestElement
+      }
+    },
+    { offset: -Infinity } // start at -Infinty so any elements offset will be above
+  ).element // returning the element itself
+}
+
+function addDragging(e) {
+  e.target.classList.add('dragging')
+}
+
+function removeDragging(e) {
+  e.target.classList.remove('dragging')
+}
+
+function updateParsedTasksAndSetLocalStorage() {
+  const arrayOfUls = [...uls]
+  for (let i = 0; i < arrayOfUls.length; i++) {
+    let category = arrayOfUls[i].classList[0].split('-tasks')[0]
+    for (let key in parsedTasks) {
+      if (key.split('-').join('') === category.split('-').join('')) {
+        const arrayOfLis = [...arrayOfUls[i].querySelectorAll('li')]
+        parsedTasks[key] = []
+        arrayOfLis.forEach((li) => {
+          parsedTasks[key].push(li.textContent)
+        })
+      }
+    }
+    localStorage.setItem('tasks', JSON.stringify(parsedTasks))
+  }
+}
+
+let sectionToPaint
+const colorPalettes = document.querySelectorAll('.color')
+colorPalettes.forEach((cp) => {
+  cp.addEventListener('click', (e) => {
+    const colorPicker = document.querySelector('#picker')
+    colorPicker.style = 'display: flex;'
+    const closestSection = e.target.closest('section')
+    sectionToPaint = closestSection
+    setTimeout(() => {
+      colorPicker.style = 'display: none;'
+    }, 5000)
+  })
+})
+
+const colorPicker = new iro.ColorPicker('#picker', {
+  width: 90,
+})
+colorPicker.on('color:change', (color) => {
+  sectionToPaint.style = `background-color: ${color.hexString}`
+})
+
+const inputs = document.querySelectorAll('input')
+inputs.forEach((input) => {
+  input.addEventListener('focus', (e) => {
+    if (e.target === searchBar) return
+    input.addEventListener('keydown', makeEnterSubmit)
+  })
+})
+
+function makeEnterSubmit(keydownEvent) {
```

Comment:
Well done for solving the problem - next time it is recommended to use html form + button submit

Reply:
Noted, thank you!